ui-tests/src/test/java/io/jenkins/plugins/analysis/warnings/WarningsPluginUiTest.java

```diff
@@ -0,0 +1,611 @@
+     */
+    @Test
+    @WithDocker
+    @WithPlugins("ssh-slaves")
+    @WithCredentials(credentialType = WithCredentials.SSH_USERNAME_PRIVATE_KEY, values = {CREDENTIALS_ID, CREDENTIALS_KEY})
+    public void shouldParseWarningsOnAgent() {
+        DumbSlave dockerAgent = createDockerAgent();
+        FreeStyleJob job = createFreeStyleJobForDockerAgent(dockerAgent, "issue_filter/checkstyle-result.xml");
+        job.addPublisher(IssuesRecorder.class, recorder -> recorder.setTool("CheckStyle", "**/checkstyle-result.xml"));
+        job.save();
+
+        Build build = buildJob(job);
+        build.open();
+
+        AnalysisSummary summary = new AnalysisSummary(build, "checkstyle");
+        assertThat(summary).isDisplayed()
+                .hasTitleText("CheckStyle: 4 warnings")
+                .hasNewSize(0)
+                .hasFixedSize(0)
+                .hasReferenceBuild(0);
+    }
+
+    private FreeStyleJob createFreeStyleJobForDockerAgent(final Slave dockerAgent, final String... resourcesToCopy) {
+        FreeStyleJob job = createFreeStyleJob(resourcesToCopy);
+        job.configure();
+        job.setLabelExpression(dockerAgent.getName());
+        return job;
+    }
+
+    /**
+     * Returns a docker container that can be used to host git repositories and which can be used as build agent. If the
+     * container is used as agent and git server, then you need to use the file protocol to access the git repository
+     * within Jenkins.
+     *
+     * @return the container
+     */
+    private DockerContainer getDockerContainer() {
+        return dockerContainer.get();
+    }
+
+    /**
+     * Creates an agent in a Docker container.
+     *
+     * @return the new agent ready for new builds
+     */
+    private DumbSlave createDockerAgent() {
+        DumbSlave agent = jenkins.slaves.create(DumbSlave.class);
+
+        agent.setExecutors(1);
+        agent.remoteFS.set("/tmp/");
+        SshSlaveLauncher launcher = agent.setLauncher(SshSlaveLauncher.class);
+
+        DockerContainer container = getDockerContainer();
+        launcher.host.set(container.ipBound(22));
+        launcher.port(container.port(22));
+        launcher.setSshHostKeyVerificationStrategy(SshSlaveLauncher.NonVerifyingKeyVerificationStrategy.class);
+        launcher.selectCredentials(CREDENTIALS_ID);
+
+        agent.save();
+
+        agent.waitUntilOnline();
+
+        assertThat(agent.isOnline()).isTrue();
+
+        return agent;
+    }
+
+    private FreeStyleJob createFreeStyleJob(final String... resourcesToCopy) {
+        FreeStyleJob job = jenkins.getJobs().create(FreeStyleJob.class);
+        ScrollerUtil.hideScrollerTabBar(driver);
+        for (String resource : resourcesToCopy) {
+            job.copyResource(WARNINGS_PLUGIN_PREFIX + resource);
+        }
+        return job;
+    }
+
+    private MavenModuleSet createMavenProject() {
+        MavenInstallation.installMaven(jenkins, DEFAULT_MAVEN_ID, "3.6.3");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'MavenInstallation.installMaven' due to existing static import 'org.jenkinsci.test.acceptance.plugins.maven.MavenInstallation.*'](https://app.codacy.com/manual/uhafner/warnings-ng-plugin/pullRequest?prid=5151082)